client/src_editor/Editor_Types.re

```diff
@@ -98,3 +103,19 @@ module Block = {
     /* Shift+Enter shortcut */
     | FcTyp_BlockExecuteAndFocusNextBlock;
 };
+
+module Link = {
+  type internalLink = {
+    revision_at: Js.Json.t,
+    sketch_id: string,
+    name: string,
+    lang,
+    code: string,
+  };
+
+  type externalLink = unit;
```

Comment:
should be an abstract type `type externalLink;` to make sure this can't be initiated